coding-concepts.js

```diff
@@ -6,49 +6,52 @@
 // Then, uncomment the console.log() and verify the output. Briefly explain why your initial answer was correct or incorrect.
 
 // --------------------INSTRUCTOR EXAMPLE: What will this log?
-const colors = ["tangerine", "magenta", "lilac", "daffodil"]
+// const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
 // a) Your answer:
 // b) Verify and explain:
 
 // --------------------1) What will this log?
 
-const cohort = "LEARN 2023"
+// const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
-
-// --------------------2) What will this log?
+// a) Your answer: 9
+// b) Verify and explain: Incorrect, I thought it was LEARN 2023
+// But I was incorrect.                               0123456789
+// I forgot .length was different to index value, so there would be no 0.
+// // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: o
+// // b) Verify and explain: Correct! The index of the string. 0 is H. Counting upwards to 4 arives at o.
 
-// --------------------3) What will this log?
+// // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer:Ruby 
+// // b) Verify and explain: Correct! the index of the array. 0 = javascript, counting to 1 = Ruby.
 
-// --------------------4) What will this log?
+// // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
+// const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// // a) Your answer: SATURDAYSUNDAY
+// // b) Verify and explain: Incorrect, toUpperCase is not a function error. It should be outside of the console log and inside brackets () in order to work. Example:
```

Comment:
The `.toUpperCase()` method is only available on strings. This code is trying to use it on an array, which is why it gives a TypeError when you try to run it.